swim/handlers_test.go

```diff
@@ -92,6 +92,13 @@ func (s *HandlerTestSuite) TestGossipStopHandler() {
 	s.True(s.testNode.node.gossip.Stopped())
 }
 
+func (s *HandlerTestSuite) TestPartitionHealerHandler() {
+	ExecuteThenWaitFor(func() {
+		_, err := s.testNode.node.partitionHealerHandler(s.ctx, &emptyArg{})
+		s.NoError(err, "calling handler should not result in error")
+	}, s.testNode.node, DiscoHealEvent{})
```

Comment:
Such use of this function might result in never ending tests in the case that this event is not emitted. Please add a timeout to this function.